java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/SearchService.java

```diff
@@ -219,10 +258,38 @@ public SearchUICore.SearchResultCollection searchCitiesByBbox(QuadRect searchBbo
         return res;
     }
     
+    public SearchUICore.SearchResultCollection searchAllPoiByBbox(QuadRect searchBbox, List<BinaryMapIndexReader> mapList) throws IOException {
+        if (!osmAndMapsService.validateAndInitConfig()) {
+            return null;
+        }
+        SearchUICore searchUICore = new SearchUICore(MapPoiTypes.getDefault(), SEARCH_LOCALE, false);
+        MapPoiTypes mapPoiTypes = searchUICore.getPoiTypes();
+        SearchCoreFactory.SearchAmenityTypesAPI searchAmenityTypesAPI = new SearchCoreFactory.SearchAmenityTypesAPI(mapPoiTypes);
+        searchUICore.registerAPI(new SearchCoreFactory.SearchAmenityByTypeAPI(mapPoiTypes, searchAmenityTypesAPI));
+        
+        SearchSettings settings = searchUICore.getPhrase().getSettings();
+        settings.setRegions(osmandRegions);
+        settings.setOfflineIndexes(mapList);
+        
+        Set<String> types = mapPoiTypes.getCategories().stream()
+                .map(PoiCategory::getKeyName)
+                .collect(Collectors.toSet());
+        SearchUICore.SearchResultCollection res = searchWithBbox(searchUICore, settings, searchBbox, types);
+        
+        int attempts = 0;
+        while ((res == null || res.getCurrentSearchResults().isEmpty()) && attempts < 10) {
```

Comment:
This is not correct! It shouldn't increase radius osm id should be exact in 10 m radius